Footer

AkshayGupta8 opened the pull request:
Created Footer with updated icon sizing.

Override LinkComponents props with `font-size` attribute updated.

Added LinkComponent prop with default argument Link in ContactLinks.jsx

AkshayGupta8 committed:
footer file

AkshayGupta8 committed:
ContactLinkes LinksComponent

AkshayGupta8 committed:
Completed Footer with Updated Links Prop

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+    </FooterOuter>
+);
+
+export default Footer;
```
This component isn't actually used anywhere. It should be at the bottom of all pages other than home.

KindaOK commented on src/components/Contact/ContactLinks.jsx:
```diff
@@ -16,7 +16,7 @@ const LinksContainer = styled.div`
 `;
 
-const Link = styled.a`
+export const Link = styled.a`
```
Don't export components from here for use in the footer. If you see that this component can be deduplicated, then extract it to `ContentStyles.js` and import it into these two components.

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+    font-weight: 700;
+`;
+
+const CustomLink = styled(Link)`
```
I don't think this is worth using inheritance for. While this is something we should improve, this is not the best place to start.

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+// import { NavLink } from "react-router-dom";
+import devices from "../utility/MediaQueries.js";
+// import logoUrl from "assets/logo-orange-cropped.png";
+// import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
```
Remove comments from imports

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import styled from "styled-components";
+// import { NavLink } from "react-router-dom";
+import devices from "../utility/MediaQueries.js";
```
Use absolute imports instead of relative imports

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import styled from "styled-components";
+// import { NavLink } from "react-router-dom";
+import devices from "../utility/MediaQueries.js";
+// import logoUrl from "assets/logo-orange-cropped.png";
+// import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+// import { SOCIAL_MEDIA_LINKS } from "utility/constants.js";
+import ContactLinks, { Link } from "./Contact/ContactLinks";
```
Include file extensions in imports

kgurushankar replied:
```suggestion
import devices from "../utility/MediaQueries.js";
import ContactLinks, { Link } from "./Contact/ContactLinks.jsx";
```

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+            <ConnectWithUsFooter>
+                CONNECT WITH US
+            </ConnectWithUsFooter>
+            <ContactLinks LinkComponent={CustomLink} />
```
This is probably a good idea given how many places we have this. However, this execution is not great. Refactor out the fragments into smaller components in a high-level file.

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+
+
+const Footer = () => (
+    <FooterOuter>
```
Reduce the amount of wrappers if possible. These seem to be redundant.

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+    `}
+`;
+
+const ConnectWithUsFooter = styled.h1`
```
There should be no more than one `<h1>` per page. I don't think this should even be a heading given that this is generic content at the bottom of each page.

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+    <FooterOuter>
+        <FooterBody>
+            <ConnectWithUsFooter>
+                CONNECT WITH US
```
The actual design seems fine. The text and element itself is fairly large though.

KindaOK reviewed (changes requested):
Needs several changes. Read [this article](https://cbea.ms/git-commit/) for writing better commit messages as well.

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+// import { SOCIAL_MEDIA_LINKS } from "utility/constants.js";
+import ContactLinks, { Link } from "./Contact/ContactLinks";
+
+const FooterOuter = styled.div`
```
The outermost wrapper should be a `<footer>`

KindaOK commented on src/components/Footer.jsx:
```diff
@@ -0,0 +1,58 @@
+import ContactLinks, { Link } from "./Contact/ContactLinks";
+
+const FooterOuter = styled.div`
+    background-color: rgba(238, 95, 48, 1);
```
Color should be hacker orange (see header)

AkshayGupta8 committed:
working on footer:
How do I add the footer component to the events page?

KindaOK committed:
Improve Footer styling on small screens

KindaOK committed:
Fix footer not on page bottom on tall screens

On either short pages or small screens, the footer would not stick
to the bottom of the screen.

Also changed it to be a <footer>.

KindaOK committed:
Add footer to remaining pages

KindaOK committed:
Add padding between footer and calendar export link

Elements were previously too close to each other.

KindaOK committed:
Merge branch 'dev' into footer
